### PR TITLE
Nithin/action group webhooks with azure active directory integration

### PR DESCRIPTION
**Pull request by nithinpnp**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by nithinpnp** [comment]
spell check error is because of specification\mediaservices\resource-manager\Microsoft.Media\stable\2018-07-01\Encoding.json file row 1200 column 95 "specified" is there. This spelling is wrong. This the reason for the failure. We don't own this resource. pls advice

**Comment by nschonni** [comment]
The spelling issue will be fixed in #6145, sorry for the noise

**Comment by nithinpnp** [comment]
@NelsonDaniel  . Could you please review this pull request ?

**Review by NelsonDaniel — COMMENTED** [review]
Looks good to me. But, requesting ARM feedback since 4 properties were added in Model "WebhookReceiver"

**Comment by nithinpnp** [comment]
I noticed that person who reviewed #WaitForArmFeedback in an earlier pull request had requested previous API version was easy diff.  I am new to the team hence did not add that in the earlier commit message. The prior API version is 2019-03-01. This set of changes is for 2019-06-01.

**Comment by nithinpnp** [comment]
@ravbhatnagar , @KrisBash  .. could you please review this ?

**Review by NelsonDaniel — APPROVED** [review]
LGTM but still waiting for ARM feedback. I will be OOF. I am adding the Reasign label in case this needs to be merged before I am back

**Comment by nithinpnp** [comment]
@shahabhijeet  Could you please review ?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json
Comment (shahabhijeet) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json]: nit: spell check

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
if ActionGroup_API.json is the new API of your REST spec, would be great if you can add a commit that shows the delta of what got changed in the new API version.
If the only thing that changed is the API version, please provide a detailed description of your PR.
Also we need ARM signOff as this is a new API version.
Having a commit that shows the diff also helps ARM to quickly review your new REST spec.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json
Comment (nithinpnp) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json]: This is not a new API. The changes are we are adding 4 not required fields to WebhookReceiver type.

```
- useAadAuth 
- objectId
- identifierUri
- tenantId 
```

Also the diff from previous version is 

 ```
 },
                "useAadAuth": {
                    "type": "boolean",
                    "default": false,
                    "description": "Indicates whether or not use AAD authentication."
                },
                "objectId": {
                    "type": "string",
                    "description": "Indicates the webhook app object Id for aad auth."
                },
                "identifierUri": {
                    "type": "string",
                    "description": "Indicates the identifier uri for aad auth."
                },
                "tenantId": {
                    "type": "string",
                    "description": "Indicates the tenant id for aad auth."
```

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json
Comment (majastrz) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json]: >CreateOrUpdate [](start = 104, length = 14)

Does the PUT of actionGroup allow partial resource body? In other words, does it act like PATCH?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json
Comment (majastrz) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json]: >null [](start = 36, length = 4)

You should set this to "nextLink" even if your service only returns 1 page. Doing so will make adding paging easier for you in the future.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json
Comment (majastrz) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json]: >null [](start = 36, length = 4)

I'd set this to "nextLink" here too even if the service only returns 1 page.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json
Comment (majastrz) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2019-06-01/actionGroups_API.json]: It's much easier for reviewers when you just make the first commit an exact copy of the previous API version. Then we can just diff commit 1 with latest and it's obvious what the delta is. In any case, I confirmed this by diffing manually and changes look good.

---
In reply to: [293928750](https://github.com/Azure/azure-rest-api-specs/pull/6218#discussion_r293928750) [](ancestors = 293928750)

**Review by majastrz — APPROVED** [review]
The incremental changes on top of 2019-03-01 look good. I added some comments that you should consider taking, but they are not blocking.

**Comment by nithinpnp** [comment]
@shahabhijeet Could you please review. Please note ARM team has signed off

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**
